### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by imran5644**
Add functionality for the buttons:

Add clickHandler to perform math operations (+, -, X, /, %, +/-)
Add style to the buttons passing the className via props
Check if the last clicked button is an operation
Use props and state to update the Virtual DOM
Use big.js to handle big numbers

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,14 +1,66 @@
...
+              total: next,
+              next: null,
+              operation: buttonName,
+            });
+          } else if (!operations.includes(lastClicked)) {
+            this.setState(calculate({ total, next, operation }, buttonName));
+          }
+          break;
+        default:
+          this.setState(calculate({ total, next, operation }, buttonName));
```
Comment (bellom) [inline src/components/App.js]: let's improve this part of the code.

If you can take a step backward, and check the instructions from the pathwright, you will see that all you need to do in this file is to create the `Constructor` and bind the `handleClick` method.
The if-else statement and switch cases should be inside `Calculate.js`

I don't think you need `lastClicked`

Example below;
```
class App extends Component {  
     constructor() {    
        super();    
        this.state = {     
            total: null, ......
        };
       this.handleClick = this.handleClick.bind(this);
    }
    
    handleClick(buttonName) {
    ......
    }
  .....
}
```
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @imran5644  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**1. Kindly check out my comments below;**
**2.** The app is broken once the user performs the following operation `10/0`, kindly write a condition to throw an error whenever a user performs the operation.
**3.** Kindly update your live demo link in your README file.

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/components/App.js [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+  }
+
+  handleClick = (buttonName) => {
+    this.setState(({
+      total, next, operation, lastClicked,
+    }) => calculate({
+      total, next, operation, lastClicked,
+    }, buttonName));
+    this.setState({ lastClicked: buttonName });
+  }
```
Comment (uche-inyama) [inline src/components/App.js]: Refactor handleClick like so:
```
 handleClick = (buttonName) => {
    this.setState(({
      total, next, operation,
    }) => calculate({
      total, next, operation,
    }, buttonName));
  }
```
Like the previous reviewer mentioned, you don't need **lastClicked**

**Review by uche-inyama — APPROVED** [review]
## Approved

### Hi Imran :slightly_smiling_face: 

I will approve this milestone. However, remove the **%** symbol in front of the result gotten, after performing a percentage operation. Also, endeavor to update **app.js**, - refer to the inline comment below.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Pull request merged**